realm/realm-library/src/main/java/io/realm/internal/OsObjectSchemaInfo.java

```diff
@@ -57,8 +58,10 @@ public Builder(String className) {
          */
         public Builder addPersistedProperty(String name, RealmFieldType type, boolean isPrimaryKey, boolean isIndexed,
                 boolean isRequired) {
-            final Property property = new Property(name, type, isPrimaryKey, isIndexed, isRequired);
-            persistedPropertyList.add(property);
+            long propertyPtr = Property.nativeCreatePersistedProperty(name,
+                    Property.convertFromRealmFieldType(type, isRequired), isPrimaryKey, isIndexed);
+            persistedPropertyPtrArray[persistedPropertyPtrCurPos] = propertyPtr;
+            persistedPropertyPtrCurPos++;
```

Comment:
why not `persistedPropertyPtrArray[persistedPropertyPtrCurPos++] = propertyPtr;`

Reply:
I'd like this way since i always have problems to understand the operator priority :P